### ISSUE TITLE
Mouse echo does not work in YouTube HTML5 player with Firefox

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#6919]
Since some months now, the mouse echo of NVDA stopped working in the HTML5 video player of YouTube when using the site with Firefox. It works OK with Internet Explorer and Edge. Even with mouse echo in NVDA enabled, NVDA will fail to read the controls, notes, subtitles and other similar stuff inside the player area. It even does not announce the appearance of an advertisement, which it did before - now it says only "alert", but not what is the alert for. Hardware acceleration in Firefox is enabled - I haven't tried to disable it to see if that will help - I'll do it now and comment on the results.

Steps to reproduce:
1. Enable the mouse echo feature in NVDA.
2. Open Firefox.
3. Go to YouTube and open a page with a video.
4. Move the mouse cursor around the controls in the player area.
5. You can open a video with added subtitls. NVDA should read them, but it does not.

Expected results:
NVDA should read the controls in the player area.

Actual results:
NVDA stays silent.

Test environment:
- Operating system: Windows 10 Pro (build 14393.693), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: next-13899,af44c1d4, in bulgarian.
- Mozilla Firefox version: 51.0.1, 32-bit, in bulgarian, hardware acceleration is enabled.
- Processor: Intel Core i5-2320 at 3.00GHz.
- RAM Memory: 8.00GB.
- Graphics: Palit GeForce GT 630, 1024MB dedicated memory + 3071MB shared system memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC662 at Intel Cougar Point PCH - High Definition Audio Controller.

**Comment by k-kolev1985** [comment]
I've disabled hardware acceleration in Firefox, restarted the browser, but the issue still persists.

**Comment by k-kolev1985** [comment]
NOTE: The issue is easier to reproduce if the player is set to full screen. After that, NVDA does not read the controls in the player area with mouse echo, even if the user exits the YouTube player from full screen mode.

**Comment by Adriani90** [comment]
In my tests NVDA reads the control type (button, button submenu etc.) but apparently does not read the label. Subtitles are not read at all.
cc: @jcsteh is this something which must be improved from Mozilla's side? In Chrome it works correctly. Even subtitles are being reported.

**Comment by Adriani90** [comment]
@k-kolev1985 how is this now on your side with NVDA 2023.1 or last alpha and Firefox 112? Please provide an update if this issue is still occuring.

**Comment by k-kolev1985** [comment]
@Adriani90 I no longer use Firefox as my default browser, though I still have it installed on my machine for test purposes. 
I can't remember what was the exact behavior of NVDA in that situation - it was 6 years ago. 
I tried it now. NVDA seems to be recognizing the controls, though it does not recognize their labels, but I'm not sure if that is a separate issue or not in this case. The subtitles are read OK now. 
I'm not sure what should we do with this case.

**Comment by jcsteh** [comment]
Not reading the labels of the controls is very likely #12047.

**Comment by Adriani90** [comment]
Closing in favor of #8337 which affects all browsers and #12067 as per comment above.
